### DISCUSSION TITLE
Audio playback crashes on iPhones

### DISCUSSION THREAD

**Discussion by luizppduarte** [tulir/whatsmeow#326] [General]
Hi,

Does anyone here suffer from audio playback problems via iphone?

When I try to play via iPhone I get the following message "This audio is no longer available. Ask the contact to resend it"

This doesn't always happen, some audios play correctly. But I couldn't see a pattern.

Android and Whatsapp Web works correctly.

Example shipping code.

```
&waProto.AudioMessage {
    Url:           proto.String(uploaded.URL),
    Mimetype:      proto.String("audio/ogg; codecs=opus"),
    FileSha256:    uploaded.FileSHA256,
    FileLength:    proto.Uint64(uploaded.FileLength),
    Seconds: 	   proto.Uint32(message.AudioTime),
    Ptt:           proto.Bool(true),
    MediaKey:      uploaded.MediaKey,
    FileEncSha256: uploaded.FileEncSHA256,
    DirectPath:    proto.String(uploaded.DirectPath),
    MediaKeyTimestamp: proto.Int64(time.Now().Unix()),
    ContextInfo:   contextWhatsapp,
    StreamingSidecar: uploaded.FileSHA256,
    Waveform: uploaded.FileSHA256,
}
```
Suggestions? 

Thanks!

**Comment by igormieski27** [comment]
Hey there! Have you found any solution or workaround? I'm going through the exact same problem. I've tried multiple different APIs, but no luck. It looks like the problem is in the recording method we are using, and not with the API itself. The error occurs only in iPhone, and doesn't follow any pattern too.

**Reply by luizppduarte** [reply]
The problem occurred when using the OGG format, changing it to MP3 worked!

**Reply by miguel-arrf** [reply]
Is this to simulate a 'voice message'? Can you show an example? Thank you!
